Doc/library/unittest.mock-examples.rst

```diff
@@ -166,6 +166,15 @@ You use the :data:`call` object to construct lists for comparing with
     >>> mock.mock_calls == expected
     True
 
+However, parameters to calls that return mocks are not recorded, which means tracking
+nested calls where the parameters used to create ancestors are important is not possible:
```

Comment:
Not sure what this means:

> the parameters used to create ancestors are important is not possible

Reply:
"tracking nested calls where the parameters used to create ancestors are important is not possible"

Reply:
OK, I see. But this can be done with `mock_calls` or `call_list` isn't it? It just cannot be all extracted from a single `call` object within `mock_calls`.

Reply:
No, it cannot, if it could, I wouldn't have reported the bug, put together a PR and spent several hours trying to figure out a way to do this that was backwards compatible ;-)

Reply:
OK, I am clearly missing something here.

```
>>> m = Mock()
>>> m.factory(important=True).deliver()
<Mock name='mock.factory().deliver()' id='139810163105576'>
>>> m.mock_calls[-1] == call.factory(important=False).deliver().call_list()
False
```

Note the false at the end. This works AFAIK.

Reply:
A call to any individual mock always returns the same mock (unless you're using a side_effect to return a different one). So you can't track the parent parameters in general as soon as there's more than one call. As noted the parameters to the original call will still be in the full call chain. I think the behaviour here is working as designed it just doesn't solve every use case.